confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -59,32 +60,30 @@ def __init__(self, url, max_schemas_per_subject=1000):
         # subj => { schema => version }
         self.subject_to_schema_versions = defaultdict(dict)
 
-    def _send_request(self, url, method='GET', body=None, headers=None):
-        if body:
-            body = json.dumps(body)
-            body = body.encode('utf8')
-        _headers = dict()
-        _headers["Accept"] = ACCEPT_HDR
+        self._session = requests.Session()
+        with self._session as s:
+            if ca_location:
+                s.verify = ca_location
+            if (cert_location or key_location) is not None:
```

Comment:
This is incorrect:
 * it first checks cert_location or key_location for truth in a () group
 * that group is returning a bool, True or False.
 * .. which you then check for `not None`

what you want to do is:
`if cert_location is not None or key_location is not None:`

Reply:
Fairly certain this is correct although it's admittedly not clear. 

The `or` operator, as I understand it, will return the first true value(evaluated left to right) in that expression as opposed to a boolean. As long as one of those two are set the the overall expression evaluates to true which is what we want. 

This is not abundantly clear though as you have rightly demonstrated so I agree it should probably be rewritten.